go-controller/pkg/metrics/ovs.go

```diff
@@ -920,3 +922,76 @@ func registerOvsMetrics(ovsDBClient libovsdbclient.Client, metricsScrapeInterval
 		go coverageShowMetricsUpdater(ovsVswitchd, stopChan)
 	})
 }
+
+type TunnelInfo struct {
+	Name       string
+	AdminState vswitchd.InterfaceAdminState
+	OperState  vswitchd.InterfaceLinkState
+	LocalIP    string
+	RemoteIP   string
+	RemoteName string
+}
+
+func listGeneveTunnels(ovsVsctl ovsClient) ([]TunnelInfo, error) {
+	var geneveTunnels []TunnelInfo
+	stdout, stderr, err := ovsVsctl("--format=json", "--columns=name,admin_state,link_state,options",
+		"find", "Interface", "type=geneve")
+	if err != nil {
+		return geneveTunnels, fmt.Errorf("failed to retrieve Geneve tunnels stderr %v: %v", stderr, err)
+	}
+	if stdout == "" {
+		return geneveTunnels, nil
+	}
+	type OvsdbData struct {
+		Data [][]interface{} `json:"data"`
+	}
+	var ovsResult OvsdbData
+	err = json.Unmarshal([]byte(stdout), &ovsResult)
+	if err != nil {
+		return geneveTunnels, fmt.Errorf("failed to parse json output for Geneve tunnels, err: %v", err)
+	}
+	for _, entry := range ovsResult.Data {
+		klog.Infof("data: %v", entry)
```

Comment:
can you give a doc text of a sample output so i can review below?

Reply:
sure, done.